Detectors/Base/src/MaterialManager.cxx

```diff
@@ -91,21 +93,66 @@ const std::unordered_map<ECut, const char*> MaterialManager::mCutIDToName = {
   {ECut::kPPCUTM, "PPCUTM"},
   {ECut::kTOFMAX, "TOFMAX"}};
 
+// Constructing a map between module names and local material density values
+void MaterialManager::createDensityMap()
+{
+  std::string token;
+  std::istringstream input(
+    o2::conf::SimMaterialParams::Instance().localDensityFactor);
+  std::vector<std::string> inputModuleNames;
+  std::vector<std::string> inputDensityValues;
+  while (std::getline(input, token, ',')) {
+    std::size_t pos = token.find(':');
+    inputModuleNames.push_back(token.substr(0, pos));
+    inputDensityValues.push_back(token.substr(pos + 1));
+  }
+  std::vector<std::string> allModuleNames = {
+    "ABSO", "CAVE", "COMP", "DIPO", "FRAME", "HALL", "MAG", "PIPE",
+    "SHIL", "CPV", "EMC", "FDD", "FT0", "FV0", "HMP", "ITS",
+    "MCH", "MFT", "MID", "PHS", "TOF", "TPC", "TRD", "ZDC",
+    "ALPIDE", "IT3", "TRK", "FT3", "A3IP"};
+  if (o2::conf::SimMaterialParams::Instance().globalDensityFactor < 0) {
+    LOG(fatal) << "Negative value "
+               << o2::conf::SimMaterialParams::Instance().globalDensityFactor
```

Comment:
I guess you could put 
`auto& globalDensityfactor = o2::conf::SimMaterialParams::Instance().globalDensityFactor;`
somewhere at the beginning of this function because you reuse it a few times.
In addition you should check its validity and `LOG(fatal)` as early as possible in this function before you do anything else

Reply:
Thank you very much Benedikt! I implemented your suggestions.